sql/src/main/java/org/apache/druid/sql/calcite/rel/Windowing.java

```diff
@@ -240,11 +240,16 @@ public static Windowing fromCalciteStuff(
 
     // Apply windowProject, if present.
     if (partialQuery.getWindowProject() != null) {
-      // We know windowProject is a mapping due to the isMapping() check in DruidRules. Check for null anyway,
-      // as defensive programming.
+      // We know windowProject is a mapping due to the isMapping() check in DruidRules.
+      // check anyway as defensive programming.
+      Preconditions.checkArgument(partialQuery.getWindowProject().isMapping());
```

Comment:
possibly extracting `partialQuery.getWindowProject()` into a local variable may make it more readable